RELEASE.md

```diff
@@ -0,0 +1,34 @@
+# Release process
```

Comment:
we already have https://github.com/kubernetes-sigs/kueue/blob/main/.github/ISSUE_TEMPLATE/NEW_RELEASE.md , so we should sync this 2 lists

Reply:
Done